### ISSUE TITLE
(Auto)Backup feature

### ISSUE THREAD

**Issue by mkae** [dmfs/opentasks#306]
It would be nice if there was a feature allowing ([scheduled] auto)backupping of all tasks.

I am asking for this because I just lost a whole lot of tasks... :-((

It is easy to loose calendar and task data if synchronisation using DAVdroid fails and one deletes the collection on the remote side which after synchronisation also dumps all local data. Once deleted I don't think I can resurrect any of it...

**Comment by dmfs** [comment]
Shouldn't backups be done server side?

**Comment by DShillito** [comment]
I would also like this feature.

Whilst I agree that, yes, backups should be done server side, it is often just easier to do it on your phone.
For example I'm currently upgrading my OwnCloud server.

I've backed up the server database, and the calenders individually, but one of them refused to save, its probably an error on my behalf, but it was good to be able to back them up from Thunderbird as a safety net, and if I could export the tasks from OpenTasks as well I would have done (belt & braces).

iCal Import/Export CalDAV allows me to backup the Calender from my phone, but does not include the tasks.

**Comment by mkae** [comment]
Yes, the possibility to back up all calendars and tasks - being originally located on different servers - just in a single backup file is a nice idea...

**Comment by Lykos153** [comment]
I agree that backups should be done on the server side. But as long as the app can be used standalone with local lists only, it should be possible to do a backup on the phone. Also, users without their own server have to rely on their service provider. They all do make backups, but only use them in case of disk failure. They won't help you after you messed with your tasks list and want a previous state back.

**Comment by Revan335** [comment]
Any news about a Backup Function?
